lms/djangoapps/django_comment_client/utils.py

```diff
@@ -819,14 +818,23 @@ def get_group_id_for_comments_service(request, course_key, commentable_id=None):
             _verify_group_exists(group_id, course_discussion_settings)
         else:
             # regular users always query with their own id.
-            group_id = get_group_id_for_user(request.user, course_discussion_settings)
+            group_id = get_group_id_for_user_from_cache(request.user, course_key)
         return group_id
     else:
         # Never pass a group_id to the comments service for a non-divided
         # commentable
         return None
 
 
+@request_cached
```

Comment:
using `request_cached` as a decorator, will it cache the call of this method until the args change? no matter how many times it is called? or it caches for any similar calls for a single request?

Reply:
Yes, it will return the cached value `for subsequent calls to the same function, with the same parameters, within a given request`

https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/request_cache/middleware.py#L68-L87